hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -480,15 +480,16 @@ Scramble enablement is done differently depending on the type of partitions.
 *  For information partitions, the scramble enablement is done on a per page basis.
    *  Software can configure for each page whether scramble is enabled.
 
-#### Flash ECC
+#### Flash ECC and ICV
 
-There are two types of flash ECC supported.
+Flash supports both ECC (error correction) and ICV (integrity check).
+While the two are used for different functions, they are implemented as two separate ECCs, thus flash supports two types of ECC.
 
-The first type is an integrity ECC used to detect whether the de-scrambled data has been modified.
-The second type is a reliability ECC used for error detection and correction on the whole flash word.
+ICV is an integrity ECC used to detect whether the de-scrambled data has been modified.
```

Comment:
Did you keep 'integrity ECC' on purpose?

Reply:
yeah. Because I want to clarify that even though it's ICV, it is implemented as an ECC.
Let me explicitly state that.

Reply:
got it